proxy/src/grpc/write.rs

```diff
@@ -3,24 +3,41 @@
 use ceresdbproto::storage::{WriteRequest, WriteResponse};
 use query_engine::executor::Executor as QueryExecutor;
 
-use crate::{error, error::build_ok_header, Context, Proxy};
+use crate::{
+    error, error::build_ok_header, grpc::metrics::GRPC_HANDLER_COUNTER_VEC, Context, Proxy,
+};
 
 impl<Q: QueryExecutor + 'static> Proxy<Q> {
     pub async fn handle_write(&self, ctx: Context, req: WriteRequest) -> WriteResponse {
         self.hotspot_recorder.inc_write_reqs(&req).await;
+
+        let mut num_rows = 0;
+        for table_request in &req.table_requests {
+            for entry in &table_request.entries {
+                num_rows += entry.field_groups.len();
+            }
+        }
+        let num_rows = num_rows as u64;
+
         match self.handle_write_internal(ctx, req).await {
             Err(e) => {
                 error!("Failed to handle write, err:{e}");
+                GRPC_HANDLER_COUNTER_VEC.write_failed.inc_by(num_rows);
```

Comment:
Maybe we can remove L20, and do casting here:
```suggestion
                GRPC_HANDLER_COUNTER_VEC.write_failed.inc_by(num_rows as u64);
```